src/lib/components/Header/Header.test.ts

```diff
@@ -0,0 +1,21 @@
+import "@testing-library/jest-dom";
+import { render, screen } from "@testing-library/svelte";
+import { describe, it, expect } from "vitest";
+
+import Header from "./Header.svelte";
+
+describe("Header", () => {
+  it("renders", () => {
+    render(Header);
+    expect(screen.getByText("<Project title>")).toBeVisible();
+  });
+  it("opens nav on menu button click and closes it on close button click", async () => {
+    render(Header, { props: { navItems: [{ id: "1", link: "/", name: "test" }] } });
+    const navContainer = screen.getByLabelText("Primary navigation");
+    expect(navContainer).not.toHaveClass("is-visible");
+    await screen.getByText("Menu").click();
```

Comment:
I think since the `Header` component is using `on:click` handlers (as opposed to some of the `on:mousedown` and `on:keydown` in the `NavItem` component), we don't actually need to use `@testing-library/user-events` here to trigger the interaction.